### ISSUE TITLE
Unsigned arithmetic in jent_delta

### ISSUE THREAD

**Issue by joshuaehill** [smuellerDD/jitterentropy-library#18]
In the code:
```
static inline uint64_t jent_delta(uint64_t prev, uint64_t next)
{
	return (prev < next) ? (next - prev) : (UINT64_MAX - prev + 1 + next);
}
```

the two branches are equivalent.

In C, arithmetic in unsigned types are done mod 1 plus the maximum value representable by that type.. See the relevant ISO for detail, but here's some relevant text from ISO/IEC 9899:2011 Section 6.2.5:

> A computation involving unsigned operands can never overflow, because a result that cannot be represented by the resulting unsigned integer type is reduced modulo the number that is one greater than the largest value that can be represented by the resulting type.

A public reference  is [here](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1548.pdf) 

Such language goes back to at least C89, so it seems likely that this behavior is consistently available.

**Comment by smuellerDD** [comment]
Am Donnerstag, 22. Oktober 2020, 19:58:25 CEST schrieb Joshua E. Hill:

Hi Joshua,

> In the code:
> ```
> static inline uint64_t jent_delta(uint64_t prev, uint64_t next)
> {
> 	return (prev < next) ? (next - prev) : (UINT64_MAX - prev + 1 + next);
> }
> ```
> 
> the two branches are equivalent.
> 
> In C, arithmetic in unsigned types are done mod 1 plus the maximum value 
representable by that type.. See the relevant ISO for detail, but here's some 
relevant text from ISO/IEC 9899:2011 Section 6.2.5:
> > A computation involving unsigned operands can never overflow, because a
> > result that cannot be represented by the resulting unsigned integer type
> > is reduced modulo the number that is one greater than the largest value
> > that can be represented by the resulting type.
> A public reference  is
> [here](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1548.pdf)
> 
> Such language goes back to at least C89, so it seems likely that this
> behavior is consistently available.

You are correct and I have that already fixed in my LRNG implementation 
(https://github.com/smuellerDD/lrng/blob/master/lrng_health.c#L332).

I will have it fixed with the next round. Thanks

Ciao
Stephan

**Comment by smuellerDD** [comment]
Fixed with 3.0.0.
